### PR TITLE
fix label and annotation for 1-line Dockerfiles

### PR DESCRIPTION
**Pull request by bbodiya-akamai**
I encountered a bug while trying to use buildah bud on 1-line Dockerfiles (i.e. just a FROM line).  There is an optimization to simply re-tag the image in this case.  However, it doesn't take into account modifications to the image that might occur from arguments passed to buildah.  In particular, --label and --annotation arguments.

I added a check for labels and annotations before opting to re-tag the existing image.  I also added tests that perform the reproduction steps below.

**Steps to reproduce the issue:**

1. Create a one-line Dockerfile with the following contents:
```
$ cat Dockerfile.from_alpine
FROM alpine
```
2. Build the image. Notice that it is simply alpine re-tagged (expected)
```
$ buildah bud -f Dockerfile.from_alpine -t from_alpine .
STEP 1: FROM alpine
Getting image source signatures
Copying blob 9d48c3bd43c5 done
Copying config 9617696764 done
Writing manifest to image destination
Storing signatures
STEP 2: COMMIT from_alpine
--> 961769676411f082461f9ef46626dd7a2d1e2b2a38e6a44364bcbecf51e66dd4

$ buildah images | grep alpine
localhost/from_alpine        latest    961769676411   2 weeks ago      5.85 MB
docker.io/library/alpine     latest    961769676411   2 weeks ago      5.85 MB
```
3. Build the image, adding a label. Notice that it is simply alpine re-tagged (**unexpected**)
```
$ buildah bud -f Dockerfile.from_alpine -t from_alpine_with_label --label test-label=true .
STEP 1: FROM alpine
STEP 2: COMMIT from_alpine_with_label
--> 961769676411f082461f9ef46626dd7a2d1e2b2a38e6a44364bcbecf51e66dd4

$ buildah images | grep alpine
localhost/from_alpine                  latest    961769676411   2 weeks ago      5.85 MB
localhost/from_alpine_with_label       latest    961769676411   2 weeks ago      5.85 MB
docker.io/library/alpine               latest    961769676411   2 weeks ago      5.85 MB
```
4. Further notice that the label was not applied to the newly built image (**unexpected**)
```
$ buildah inspect localhost/from_alpine_with_label | grep Labels\"
            "Labels": null
            "Labels": null
```
5. Note: the same occurs with --annotation as with --label.

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
bot, add author to whitelist

**Comment by TomSweeneyRedHat** [comment]
LGTM
@ bbodiya-akamai TYVM for spotting the error and providing the PR to fix it!
The CI test issues are not at all related to this PR.  We're investigating the issue now and we're hoping to have it corrected in the not too distant future.

**Comment by rhatdan** [comment]
@bbodiya-akamai The travis issues seems to be your tests blowing up.

**Comment by TomSweeneyRedHat** [comment]
Oh shoot, you're right @rhatdan.  I just looked at the Dawrin CI error which is not related.

File: tests/bud.bats [hunk 1]
```diff
@@ -1641,3 +1641,34 @@ load helpers
...
+
+  buildah rmi ${target}
+}
+
+@test "bud-no-change-label" {
+  parent=alpine
+  target=no-change-image
+  buildah bud --label "test=label" --signature-policy ${TESTSDIR}/policy.json -t ${target} ${TESTSDIR}/bud/no-change
+  run_buildah --log-level=error inspect --format '{{printf "%q" .Docker.Config.Labels}}' ${target}
+  expect_output 'map["test":"label"]'
```
Comment (TomSweeneyRedHat) [inline tests/bud.bats]: I suspect this will work now.  If by chance it doesn't try:
`  [[ "$output" == 'map["test":"label"]' ]]`

Similar suggestion below

**Comment by bbodiya-akamai** [comment]
I missed the deprecation of --debug after I forked, sorry about that.  Seems to be passing now.

**Comment by rhatdan** [comment]
LGTM
Darwin seems to be hosed up, Forcing merge.

**Pull request merged**